app/models/article.rb

```diff
@@ -585,22 +585,12 @@ def create_password
 
   def update_cached_user
     if organization
-      self.cached_organization = OpenStruct.new(set_cached_object(organization))
+      self.cached_organization = Articles::CachedEntity.from_object(organization)
```

Comment:
We now have a simple domain model with a speclalized constructor for turning anything with the right attributes (including the currently saved ostructs) into the structs that get cached.